bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -295,10 +302,20 @@ public String authority() {
    * {@link ClientCall#start(ClientCall.Listener, io.grpc.Metadata)} is invoked.
    */
   @Override
-  public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(
-      MethodDescriptor<ReqT, RespT> methodDescriptor, CallOptions callOptions) {
+  public <ReqT, RespT> ClientCall<ReqT, RespT>
+      newCall(final MethodDescriptor<ReqT, RespT> methodDescriptor, final CallOptions callOptions) {
     Preconditions.checkState(!shutdown, "Cannot perform operations on a closed connection");
-    return getNextChannel().newCall(methodDescriptor, callOptions);
+    return new CheckedForwardingClientCall<ReqT, RespT>(
+        getNextChannel().newCall(methodDescriptor, callOptions)) {
+      @Override
+      protected void checkedStart(ClientCall.Listener<RespT> listener, Metadata headers)
+          throws Exception {
+        for (HeaderInterceptor interceptor : headerInterceptors) {
+          interceptor.updateHeaders(headers);
+        }
+        delegate().start(listener, headers);
+      }
+    };
```

Comment:
This is a pretty small class.  We don't have a separate class for the `InstrumentedChannel`'s `CheckedForwardingClientCall`.  I don't have a strong feeling about this either way.